connectivity/FEATURE_BLE/source/cordio/source/PalGapImpl.cpp

```diff
@@ -683,16 +683,34 @@ void PalGap::gap_handler(const wsfMsgHdr_t *msg)
         }   break;
 #endif //  BLE_FEATURE_EXTENDED_ADVERTISING && BLE_ROLE_BROADCASTER
 
-#if BLE_FEATURE_EXTENDED_ADVERTISING && BLE_ROLE_OBSERVER
+#if BLE_ROLE_OBSERVER
+        case DM_SCAN_START_IND:
+            if (!handler) { break; }
+            handler->on_scan_started(msg->status == HCI_SUCCESS);
+            break;
+
+        case DM_SCAN_STOP_IND:
+            if (!handler) { break; }
+            handler->on_scan_stopped(msg->status == HCI_SUCCESS);
+            break;
+
+#if BLE_FEATURE_EXTENDED_ADVERTISING
+        case DM_EXT_SCAN_START_IND:
+            if (!handler) { break; }
+            handler->on_scan_started(msg->status == HCI_SUCCESS);
+            break;
+
         case DM_EXT_SCAN_STOP_IND: {
-            if (!handler) {
-                break;
+            if (!handler) { break; }
+
+            if (get_gap().ext_scan_stopping) {
+                get_gap().ext_scan_stopping = false;
```

Comment:
this should be set even if there is not handler, so the handler check above needs to be done later